wagit/headers/GitExceptions.h

```diff
@@ -11,26 +12,37 @@
 
 namespace WAGit::Exceptions
 {
-    struct NotAGitRepositoryException : public std::exception {
-        std::string exceptionMessage;
+    namespace FS = std::filesystem;
 
-        explicit NotAGitRepositoryException(const std::string& path)
-                : exceptionMessage("Not a Git repository " + path)
+    using Exception = std::exception;
+    using Path = FS::path;
+    using String = std::string;
```

Comment:
Why can't we just using namespace std instead of doing this for every single class?

Reply:
In general, doing a use namespace to bring in everything is bad. Basically, imagine you have a function `foo` in namespace `F`, you have another function `foo` in namespace `G`. See the problem?
`namespace std` is huge!